examples/hooks-e-commerce/App.tsx

```diff
@@ -0,0 +1,369 @@
+import algoliasearch from 'algoliasearch/lite';
+import React, { useRef } from 'react';
+import {
+  ClearRefinements,
+  Configure,
+  HierarchicalMenu,
+  Hits,
+  HitsPerPage,
+  InstantSearch,
+  Pagination,
+  RefinementList,
+  SearchBox,
+  SortBy,
+  ToggleRefinement,
+  Highlight,
+  Snippet,
+} from 'react-instantsearch-hooks-web';
+
+import {
+  AlgoliaSvg,
+  ClearFiltersMobile,
+  NoResults,
+  Panel,
+  PriceSlider,
+  Ratings,
+  ResultsNumberMobile,
+  SaveFiltersMobile
+} from './components';
+
+import './Theme.css';
+import './App.css';
+import './App.mobile.css';
+import { formatNumber } from './utils';
+
+const searchClient = algoliasearch(
+  'latency',
+  '6be0576ff61c053d5f9a3225e2a90f76'
+);
+
+export function App() {
+  const containerRef = useRef<HTMLElement>(null);
+  const headerRef = useRef(null);
+
+  function openFilters() {
+    document.body.classList.add('filtering');
+    window.scrollTo(0, 0);
+    window.addEventListener('keyup', onKeyUp);
+    window.addEventListener('click', onClick);
+  }
+
+  function closeFilters() {
+    document.body.classList.remove('filtering');
+    containerRef.current!.scrollIntoView();
+    window.removeEventListener('keyup', onKeyUp);
+    window.removeEventListener('click', onClick);
+  }
+
+  function onKeyUp(event) {
+    if (event.key !== 'Escape') {
+      return;
+    }
+
+    closeFilters();
+  }
+
+  function onClick(event) {
+    if (event.target !== headerRef.current) {
+      return;
+    }
+
+    closeFilters();
+  }
+
+  return (
+    <InstantSearch
+      searchClient={searchClient}
+      indexName="instant_search"
+      routing={true}
+    >
+      <header className="header" ref={headerRef}>
+        <p className="header-logo">
+          <AlgoliaSvg />
+        </p>
+
+        <p className="header-title">Stop looking for an item — find it.</p>
+
+        <SearchBox
+          placeholder="Product, brand, color, …"
+          submitIconComponent={SubmitIcon}
+        />
+      </header>
+
+      <Configure
+        attributesToSnippet={['description:10']}
+        snippetEllipsisText="…"
+        removeWordsIfNoResults="allOptional"
+      />
+
+      <main className="container" ref={containerRef}>
+        <div className="container-wrapper">
+          <section className="container-filters" onKeyUp={onKeyUp}>
+            <div className="container-header">
+              <h2>Filters</h2>
+
+              <div className="clear-filters" data-layout="desktop">
+                <ClearRefinements
+                // translations={{
```

Comment:
this could be turned in a custom widget, also used for no results (currently that's inline)